pkg/api/validation/validation.go

```diff
@@ -455,6 +459,25 @@ func validateGlusterfs(glusterfs *api.GlusterfsVolumeSource) errs.ValidationErro
 	return allErrs
 }
 
+var validMetadataFieldPathExpressions = util.NewStringSet("metadata.name", "metadata.namespace", "metadata.labels", "metadata.annotations")
+
+func validateMetadataVolumeSource(metadata *api.MetadataVolumeSource) errs.ValidationErrorList {
+	allErrs := errs.ValidationErrorList{}
+	for _, metadataFile := range metadata.Items {
+		if len(metadataFile.Name) == 0 {
+			allErrs = append(allErrs, errs.NewFieldRequired("name"))
+		}
+		if path.IsAbs(metadataFile.Name) {
+			allErrs = append(allErrs, errs.NewFieldForbidden("name", "must not be an absolute path"))
+		}
+		if strings.Contains(metadataFile.Name, "..") {
```

Comment:
shouldn't you split and check all pieces for == ".." ?  Otherwsie foo..bar (which is valid) will fail.